### DISCUSSION TITLE
`Translations` Metadata Decomposition

### DISCUSSION THREAD

**Discussion by quinnmcphail** [forcedotcom/cli#3014] [Ideas]
I've been watching the [CustomLabels](https://github.com/forcedotcom/cli/discussions/2948) and [PermissionSets](https://github.com/forcedotcom/cli/discussions/2993) decomposition discussions, and a similar treatment to the [Translations metadata type](https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_translations.htm) would be welcome.

Taking it a step further, moving the translation metadata near their respective parent metadata components would be ideal. For instance, with `CustomLabels`, having the translation metadata in the same folder structure as the `CustomLabel` itself would allow easier maintenance and allow missing translations to be easily identified.

**Comment by VivekMChawla** [comment]
@quinnmcphail - Thanks for opening this discussion.

We plan to work on Profiles and Translations next, likely starting sometime after mid-October.

When the time comes, I'll share design proposals with the community for feedback.

In the meantime, could you share more about how you'd like to see `CustomLabels` and `Translations` co-exist?  

Specifically, could you take the file structure example below and insert decomposed `Translations` in the way that you'd find most helpful?

```
sfdx-project 
 ├─ force-app
 │   └─ main
 │       └─ default
 │           └─ labels
 │               ├─ Login_Message.label-meta.xml  
 │               └─ Logout_Message.label-meta.xml
 └─ otherPackageDir
     └─ labels
         ├─ Error
         │   └─ Authentication_Error.label-meta.xml
         └─ Warning
             └─ Timeout_Warning.label-meta.xml
```

**Reply by quinnmcphail** [reply]
@VivekMChawla - Sure! I think this would probably be the most helpful. I'm only showing an example of the `Login_Message` label with translations:

```
sfdx-project 
 ├─ force-app
 │   └─ main
 │       └─ default
 │           └─ labels
 │               ├─ Login_Message
 │               │   ├─ Login_Message.label-meta.xml
 │               │   ├─ Login_Message.de.translation-meta.xml
 │               │   └─ Login_Message.fr.translation-meta.xml
 │               └─ Logout_Message.label-meta.xml
 └─ otherPackageDir
     └─ labels
         ├─ Error
         │   └─ Authentication_Error.label-meta.xml
         └─ Warning
             └─ Timeout_Warning.label-meta.xml
```

**Comment by thvd** [comment]
So right now it's not supported to have two translations files in separate package dirs? Because now if you deploy two translations in one operation, it only deploys the second. 
Sample: 


```bash

sf project deploy start \
  -d force-app/main/default/translations \
  -d cpq/main/default/translations
  -o myorg \
  -w 60

```


Would be nice if this new feature would solve this limitation.

**Comment by mcarvin8** [comment]
If you'd like to decompose sooner while waiting for an official Salesforce beta, feel free to use my Salesforce CLI plugin - https://github.com/mcarvin8/sf-decomposer

Supports decomposing a good junk of metadata types available in Salesforce's metadata registry. 

This allows you to also be selective with what metadata you want to decompose. Salesforce's decomposition is all or nothing, meaning if you want to decompose workflows, all your workflows will be decomposed. My plugin lets you create an ignore file to pick and choose what files you would want decomposed under particular metadata types.

**Comment by markoc-geckohrm** [comment]
Will this feature be supported in near future?
